Sending an invalid bearer token causes a 500 error

See logs below.

Sending a non-sensical value such as "Authorization: bearer abcd" should result in the server responding with `401 Unauthorized`.

Logs:
```
[2024-08-13 12:58:35 +0000] [88] [ERROR] Exception occurred while handling uri: 'https://renku-ci-gw-710.dev.renku.ch/api/data/user'
Traceback (most recent call last):
  File "/app/env/lib/python3.12/site-packages/jwt/api_jws.py", line 257, in _load
    signing_input, crypto_segment = jwt.rsplit(b".", 1)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected 2, got 1)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "handle_request", line 102, in handle_request
    try:
         
  File "/app/env/lib/python3.12/site-packages/renku_data_services/base_api/auth.py", line 37, in decorated_function
    user = await authenticator.authenticate(token, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/renku_data_services/authn/keycloak.py", line 67, in authenticate
    parsed = self._validate(access_token)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/renku_data_services/authn/keycloak.py", line 50, in _validate
    sk = self.jwks.get_signing_key_from_jwt(token)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/jwt/jwks_client.py", line 111, in get_signing_key_from_jwt
    unverified = decode_token(token, options={"verify_signature": False})
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/jwt/api_jwt.py", line 151, in decode_complete
    decoded = api_jws.decode_complete(
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/jwt/api_jws.py", line 198, in decode_complete
    payload, signing_input, header, signature = self._load(jwt)
                                                ^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/jwt/api_jws.py", line 260, in _load
    raise DecodeError("Not enough segments") from err
jwt.exceptions.DecodeError: Not enough segments
[2024-08-13 12:58:48 +0000] [88] [ERROR] Exception occurred while handling uri: 'https://renku-ci-gw-710.dev.renku.ch/api/data/user'
Traceback (most recent call last):
  File "/app/env/lib/python3.12/site-packages/jwt/api_jws.py", line 257, in _load
    signing_input, crypto_segment = jwt.rsplit(b".", 1)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected 2, got 1)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "handle_request", line 102, in handle_request
    try:
         
  File "/app/env/lib/python3.12/site-packages/renku_data_services/base_api/auth.py", line 37, in decorated_function
    user = await authenticator.authenticate(token, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/renku_data_services/authn/keycloak.py", line 67, in authenticate
    parsed = self._validate(access_token)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/renku_data_services/authn/keycloak.py", line 50, in _validate
    sk = self.jwks.get_signing_key_from_jwt(token)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/jwt/jwks_client.py", line 111, in get_signing_key_from_jwt
    unverified = decode_token(token, options={"verify_signature": False})
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/jwt/api_jwt.py", line 151, in decode_complete
    decoded = api_jws.decode_complete(
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/jwt/api_jws.py", line 198, in decode_complete
    payload, signing_input, header, signature = self._load(jwt)
                                                ^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.12/site-packages/jwt/api_jws.py", line 260, in _load
    raise DecodeError("Not enough segments") from err
jwt.exceptions.DecodeError: Not enough segments
```

--- a/components/renku_data_services/base_api/error_handler.py
+++ b/components/renku_data_services/base_api/error_handler.py
@@ -7,6 +7,7 @@ from collections.abc import Mapping, Set
 from sqlite3 import Error as SqliteError
 from typing import Any, Optional, Protocol, TypeVar, Union
 
+import jwt
 from asyncpg import exceptions as postgres_exceptions
 from pydantic import ValidationError as PydanticValidationError
 from sanic import HTTPResponse, Request, SanicException, json
@@ -129,6 +130,8 @@ class CustomErrorHandler(ErrorHandler):
                 formatted_exception = errors.ValidationError(
                     message="The provided input is too large to be stored in the database"
                 )
+            case jwt.exceptions.InvalidTokenError():
+                formatted_exception = errors.InvalidTokenError()
         self.log(request, formatted_exception)
         if formatted_exception.status_code == 500 and "PYTEST_CURRENT_TEST" in os.environ:
             # TODO: Figure out how to do logging properly in here, I could not get the sanic logs to show up from here

--- a/components/renku_data_services/errors/errors.py
+++ b/components/renku_data_services/errors/errors.py
@@ -46,6 +46,13 @@ class UnauthorizedError(BaseError):
 
 
 @dataclass
+class InvalidTokenError(UnauthorizedError):
+    """The supplied jwt is invalid."""
+
+    message: str = "The supplied credentials (jwt) are not valid."
+
+
+@dataclass
 class ForbiddenError(BaseError):
     """Raised when the provided credentials do not grant permission for the current operation."""
 

